### PR TITLE
V3-1108 Satellite console forgot password backend

### PR DESCRIPTION
**Pull request by crawter**

### CODE REVIEW

File: satellite/console/service.go [hunk 1]
```diff
@@ -229,30 +192,53 @@ func (s *Service) UpdateAccount(ctx context.Context, info UserInfo) (err error)
 	})
 }
 
-// ChangePassword updates password for a given user
-func (s *Service) ChangePassword(ctx context.Context, pass, newPass string) (err error) {
+// RequestChangePassword - sends email with link to restore forgotten password
```
Comment (egonelbre) [inline satellite/console/service.go]: nit, remove dash from comments. https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

File: satellite/console/service.go [hunk 2]
```diff
@@ -613,6 +599,19 @@ func (s *Service) Authorize(ctx context.Context) (a Authorization, err error) {
 	}, nil
 }
 
+// GenerateToken - is a method for generating token with users info
+func (s *Service) GenerateToken(ctx context.Context, id uuid.UUID, email string, expirationDate time.Time) (token string, err error) {
```
Comment (rikysya) [inline satellite/console/service.go]: I am not sure that token configuration should be dried out

File: satellite/console/service.go [hunk 3]
```diff
@@ -664,6 +663,36 @@ func (s *Service) authorize(ctx context.Context, claims *consoleauth.Claims) (*U
 	return user, nil
 }
 
+// getUserFromClaims - returns User from token
```
Comment (rikysya) [inline satellite/console/service.go]: fix comment

File: satellite/console/service.go [hunk 4]
```diff
@@ -664,6 +663,36 @@ func (s *Service) authorize(ctx context.Context, claims *consoleauth.Claims) (*U
...
+func (s *Service) getUserFromToken(ctx context.Context, token consoleauth.Token) (*User, error) {
```
Comment (rikysya) [inline satellite/console/service.go]: Not sure it should be dried out

File: satellite/console/service.go [hunk 5]
```diff
@@ -229,30 +192,53 @@ func (s *Service) UpdateAccount(ctx context.Context, info UserInfo) (err error)
...
+// ChangePasswordExternal updates password for a given user in reset password flow
+func (s *Service) ChangePasswordExternal(ctx context.Context, resetPasswordToken, pass, newPass string) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	token, err := consoleauth.FromBase64URLString(resetPasswordToken)
+	if err != nil {
+		return
+	}
+
+	user, err := s.getUserFromToken(ctx, token)
```
Comment (rikysya) [inline satellite/console/service.go]: This should not skip checking token expiration time. Reset token can go through the same authenticate flow as authorization token. As we semantically defined it as separate self-encoded token type, may be should not dry out it's authorizing.

**Comment by mniewrzal** [comment]
@crawter do you plan to continue developing this PR?

**Comment by crawter** [comment]
Closed, because this task was reimplemented be @Barterio 
https://github.com/storj/storj/pull/1665
https://github.com/storj/storj/pull/1916

**Pull request closed without merging**